client/isolated.go

```diff
@@ -0,0 +1,124 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary"
+	"github.com/docker/notary/cryptoservice"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+	"io"
+	"io/ioutil"
+)
+
+// AddTargetToFile adds the target to the file found at filePath, and writes the updated and re-signed version
+// to the outBuf.
+// If signingKeys are provided, they are used to sign the file. If lookupKeys are provided, they
+// will be looked up in the CryptoService and used for signing where found.
+// If _both_ signingKeys, and lookupKeys are provided, the public keys in lookupKeys will be added to signingKeys
+// and the merged list will be used for signing.
+// If _neither_ signingKeys, no lookupKeys are provided, the key IDs associated with any existing signatures are
+// used to search for keys in the CryptoService and used to sign.
+func AddTargetToFile(baseDir string, retriever notary.PassRetriever, outBuf io.Writer, filePath string, target *Target, signingKeys map[string]data.PrivateKey, lookupKeys map[string]data.PublicKey) error {
+	// do some setup
+	var fakeRole data.RoleName = data.CanonicalTargetsRole
+	ks, err := getKeyStores(baseDir, retriever)
+	if err != nil {
+		return err
+	}
+	cs := cryptoservice.NewCryptoService(ks...)
+
+	var parsed *data.SignedTargets
+	if filePath == "-" {
```

Comment:
Hmmm, I think I'd probably have to switch to a `*string` then to differentiate "no value" (i.e. use traditional `notary add` logic) vs "empty value" (use new `notary add` logic, creating new empty targets file). Will take a look.